### ISSUE TITLE
Rate limit token does not work on GHE

### ISSUE THREAD

**Issue by mvs5465** [actions/setup-python#638]
**Description:**
Using the `token` parameter for the Github Action, the token is not used. 

**Action version:**
v4

**Platform:**
- [x] Ubuntu

**Runner type:**
- [x] Self-hosted

**Tools version:**
3.x

**Repro steps:**  
Add a Github Action workflow like this to your Github Enterprise repository:

```
      - name: Install Python
        uses: actions/setup-python@v4
        with:
          token: ${{ secrets.DOTCOM_TOKEN }}
          python-version: '3.x'
```

Then also add https://github.com/actions/setup-python/pull/510#discussion_r982394709 using the same token variable.

We can tell by the rate-limit response that the token has not been used. 

**Expected behavior:**

Expect action to use token.

**Actual behavior:**

`API rate limit exceeded for <ip>`

We confirmed using the rate limit query that the rate limit quotas have not changed, and thus the token has not been used by the action.

**Comment by e-korolevskii** [comment]
Hello @mvs5465 
Thank you for your report. We'll investigate the issue and get back to you.

**Comment by jay0129** [comment]
I got the same problem.
I downgraded action version to `actions/setup-python@v3.1.2` and it successfully uses `token`.

**Comment by chebong-aus** [comment]
Hi @e-korolevskii, are there any updates on this issue? I am facing the exact same issue in my GitHub Enterprise repository.

**Comment by dmitry-shibanov** [comment]
Hello everyone. Could you please take a look at this [issue](https://github.com/actions/setup-python/issues/683) because I think they're related.

**Comment by chebong-aus** [comment]
Hi @dmitry-shibanov, looking at the link you provided, not sure if they are related because the error I get is `Error: API rate limit exceeded for x. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)`. This is while passing in a GitHub.com Token.

**Comment by priya-kinthali** [comment]
Hello @mvs5465 👋,
Could you kindly verify if the issue related to the rate limit has been resolved and confirm that no other issues have emerged while using the [setup-actions-demo/setup-python@main branch](https://github.com/setup-actions-demo/setup-python).

**Comment by priya-kinthali** [comment]
Hello @mvs5465👋,
Just a gentle reminder, Could you kindly verify if the issue related to the rate limit has been resolved and confirm that no other issues have emerged while using the [setup-actions-demo/setup-python@main branch](https://github.com/setup-actions-demo/setup-python).

**Comment by priya-kinthali** [comment]
Hello @mvs5465 👋,
As part of the latest updates, the potential improvement to leverage the raw API for retrieving version-manifest has been implemented and released in [v5.1.0](https://github.com/actions/setup-python/releases/tag/v5.1.0). This new approach should mitigate the rate limit exhaustion without requiring an additional GitHub token.
Please update to the latest version and let us know if you encounter any further issues. As this update resolves the reported problem, we will close this issue.
Thank you!
